server/src/main/java/com/linecorp/centraldogma/server/internal/storage/project/DefaultProject.java

```diff
@@ -132,13 +147,24 @@ public class DefaultProject implements Project {
         name = rootDir.getName();
         repos = newRepoManager(rootDir, repositoryWorker, purgeWorker, cache);
 
+        final boolean useDogmaRepoAsMetaRepo;
+        if (dogmaProject == null) {
+            useDogmaRepoAsMetaRepo = false;
+        } else {
+            final Repository dogmaProjectDogmaRepository = dogmaProject.repos().get(REPO_DOGMA);
+            final Entry<JsonNode> entry = dogmaProjectDogmaRepository.getOrNull(
+                    Revision.HEAD, Query.ofJson(META_TO_DOGMA_MIGRATION_JOB)).join();
+            useDogmaRepoAsMetaRepo = entry != null;
```

Comment:
If you intended to refresh `MetaRepository`, adding a new command to propagate the event seems like a complex approach. Instead, it might be simpler to add a property such as `{ "metaRepository: <repoName> }` in `metadata.json` to indicate which repository is used for `MetaRepository` and handle `resetMetaRepository()` logic inside `RepositoryListener`.
https://github.com/line/centraldogma/blob/920d7b115b5c27c9ea313eb3b0025606cf4b5a6f/server/src/main/java/com/linecorp/centraldogma/server/internal/storage/project/DefaultProject.java#L262

Reply:
Thanks for the suggestion.
Here's the migration flow I'm planning:
- Set the project's Dogma repository to read-only
- Migrate data from the meta repository to the Dogma repository
- Add a /meta-to-dogma-migrated.json file to the Dogma repository
- Reset the meta repository on all replicas via the command
- Set the project's Dogma repository to active
- Proceed to the next project

The migration is a one-time operation, and the related logic will be removed afterward.
I tried to avoid polluting the existing logic as much as possible, and I felt it would be more intuitive for the leader to handle both the migration and status transition.

Reply:
I see. As the code is only used during migration, I think it’s fine to leave it as is without spending extra effort.